pallets/manta-sbt/src/lib.rs

```diff
@@ -602,6 +821,151 @@ where
             }
         })
     }
+
+    /// Checks that post is `ToPrivate` with a value of one
+    #[inline]
+    fn check_post_shape(post: &TransferPost, asset_id: StandardAssetId) -> DispatchResult {
+        // Checks that it is indeed a to_private post with a value of 1
+        ensure!(
+            post.sources.len() == 1
+                && post.sender_posts.is_empty()
+                && post.receiver_posts.len() == 1
+                && post.sinks.is_empty(),
+            Error::<T>::InvalidShape
+        );
+        // Checks that value is one, this is defensive as value is not used for SBT.
+        ensure!(
+            post.sources.first() == Some(&ENCODED_ONE),
+            Error::<T>::ValueNotOne
+        );
+
+        let post_id: StandardAssetId = post
+            .asset_id
+            .and_then(id_from_field)
+            .ok_or(Error::<T>::InvalidAssetId)?;
+        // Ensure asset id is correct, only a single unique asset_id mapped to account is valid
+        ensure!(asset_id == post_id, Error::<T>::InvalidAssetId);
+        Ok(())
+    }
+
+    /// Checks that signature was generated using the `TransferPost` proof field as payload
+    #[inline]
+    fn verify_eip712_signature(proof: &Proof, sig: &[u8; 65]) -> Option<EvmAddress> {
+        let msg = Self::eip712_signable_message(proof);
+        let msg_hash = keccak_256(msg.as_slice());
+
+        recover_signer(sig, &msg_hash)
+    }
+
+    /// Eip-712 message to be signed
+    #[inline]
+    fn eip712_signable_message(proof: &Proof) -> Vec<u8> {
```

Comment:
Seems we can use a fixed size of array here.

Reply:
Not sure there is an easy way to avoid allocating the vec.

Could do this but in practice I don't think there is any tangible benefit. Maybe you could share a snippit if you have an idea of best way to do it:

```rust
    /// Eip-712 message to be signed
    #[inline]
    fn eip712_signable_message(proof: &Proof) -> [u8; 66] {
        let domain_separator = Self::evm_account_domain_separator();
        let payload_hash = Self::evm_account_payload_hash(proof);

        let mut msg = b"\x19\x01".to_vec();
        msg.extend_from_slice(&domain_separator);
        msg.extend_from_slice(&payload_hash);
        msg.try_into().expect("bla")
    }
```

Reply:
Agreed, there's almost no benefit we can get.